godep/client.go

```diff
@@ -68,24 +78,62 @@ type Client struct {
 	store ClientStorage
 
 	// an HTTP client that handles DEP API authentication and session management
-	client *http.Client
+	client  depclient.Doer
+	errHook func(ctx context.Context, err error) error
+}
+
+// ClientOption defines the functional options type for NewClient.
+type ClientOption func(*Client)
+
+// WithMiddleware applies a middleware function to the client's HTTP client.
+// It can be useful to transform the HTTP request just before it is sent, or
+// to intercept the HTTP response and error before they are returned to the
+// caller.
+func WithMiddleware(mw func(depclient.Doer) depclient.Doer) ClientOption {
```

Comment:
This seems unused in https://github.com/fleetdm/fleet/pull/9091. Typically, if you want to intercept the request/response you can use an `*http.Client` with a custom `Transport`.

Reply:
Yeah it's not, I did attempt that approach first but that does not work because the translation to HTTPError/AuthError happens after the Do call. As I had it already implemented, and it can be a useful extension in general (transport-level is often too low-level), I left it there but yeah I don't mind removing it.

Reply:
I removed it.